librz/core/cbin.c

```diff
@@ -2756,11 +2780,14 @@ static int bin_sections(RzCore *r, PJ *pj, int mode, ut64 laddr, int va, ut64 at
 			if (print_segments != s->is_segment) {
 				continue;
 			}
-			RzInterval pitv = (RzInterval){ s->paddr, s->size };
+			RzInterval pitv = (RzInterval){ s->paddr, s->size }; // 'iSS' command
 			RzInterval vitv = (RzInterval){ s->vaddr, s->vsize };
+
 			rz_num_units(humansz, sizeof(humansz), s->size);
 			RzListInfo *info = rz_listinfo_new(s->name, pitv, vitv, s->perm, strdup(humansz));
 			rz_list_append(list, info);
+			rz_list_append(list, rz_str_newf("%lld", s->type));
```

Comment:
1) The RzList is expecting only `RzListInfo`s to be added to it so you cannot add a string. (Segfaults when you run `iS=`)

The types and flags could probably be added on to the `strdup(humansz)` if we want this.

This command (`iS=`) is undocumented and I don't know if anyone actually uses this. Thus it might not be worth adding types and flags to this command, but I want to let the maintainers make the call on this.

2) Dont use "%lld" as it is non portable. Use the `PFMT64` macros as mentioned in [Developers.md](https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md)

Reply:
Nevermind about not adding types and flags to `iS=`. It is documented

Reply:
@PinkNoize in general we are using custom formats like `PFMT64d` since `%lld` is not always portable.